eth/backend.go

```diff
@@ -97,8 +97,8 @@ type Ethereum struct {
 
 	networkID     uint64
 	netRPCService *ethapi.PublicNetAPI
-
-	lock sync.RWMutex // Protects the variadic fields (e.g. gas price and etherbase)
+	NodeID        string       // @rayyuan id of node
```

Comment:
As we discussed, you can have the client get the node ID from its tracking of the peer rather having t  
o store it here.